nixup/lib/eval-config.nix

```diff
@@ -0,0 +1,58 @@
+# From an end-user configuration file (`configuration'), build a NixUP
+# configuration object (`config') from which we can retrieve option
+# values.
+
+# !!! Please think twice before adding to this argument list!
+# Ideally eval-config.nix would be an extremely thin wrapper
+# around lib.evalModules, so that modular systems that have nixos configs
+# as subcomponents (e.g. the container feature, or nixops if network
+# expressions are ever made modular at the top level) can just use
+# types.submodule instead of using eval-config.nix
+{ # !!! system can be set modularly, would be nice to remove
```

Comment:
can these first three (system, pkgs, baseModules) already be addressed (which I guess would mean removed, based on the comments?)